workflow/howto/vercel-ai-sdk.mdx

```diff
@@ -0,0 +1,261 @@
+
+AI SDKs (Vercel AI SDK, OpenAI SDK etc.) uses the client's default fetch implementation to make API requests, but allows you to provide a custom fetch implementation.
+
+In the case of Upstash Workflow, we need to use the `context.call` method to make HTTP requests. We can create a custom fetch implementation that uses `context.call` to make requests. By using `context.call`, Upstash Workflow is the one making the HTTP request and waiting for the response, even if it takes too long to receive response from the LLM.
+
+The following code snippet can also be generalized to work with other LLM SDKs, such as Anthropic or Google.
+
+```typescript {18-24}
+import { createOpenAI } from '@ai-sdk/openai';
+import { WorkflowAbort, WorkflowContext } from '@upstash/workflow';
+import { HTTPMethods } from '@upstash/qstash';
+
+export const createWorkflowOpenAI = (context: WorkflowContext) => {
+  return createOpenAI({
+    compatibility: "strict",
+    fetch: async (input, init) => {
+      try {
+        // Prepare headers from init.headers
+        const headers = init?.headers
+          ? Object.fromEntries(new Headers(init.headers).entries())
+          : {};
+
+        // Prepare body from init.body
+        const body = init?.body ? JSON.parse(init.body as string) : undefined;
+
+        // Call the workflow context
+        const responseInfo = await context.call("call step", {
+          url: input.toString(),
+          method: init?.method as HTTPMethods,
+          headers,
+          body,
+        });
+
+        // Construct headers for the response
+        const responseHeaders = new Headers(
+          Object.entries(responseInfo.header).reduce((acc, [key, values]) => {
+            acc[key] = values.join(", ");
+            return acc;
+          }, {} as Record<string, string>)
+        );
+
+        // Return the constructed response
+        return new Response(JSON.stringify(responseInfo.body), {
+          status: responseInfo.status,
+          headers: responseHeaders,
+        });
+      } catch (error) {
+        if (error instanceof WorkflowAbort) {
+          throw error
+        } else {
+          console.error("Error in fetch implementation:", error);
+          throw error; // Rethrow error for further handling
+        }
+      }
+    },
+  });
+};
+```
+
+### Using OpenAI client to generate text
+
+Now that we've created the OpenAI client, we can use it to generate text or embeddings etc.
+
+For that, we're going to create a new workflow endpoint that uses the payload as prompt to generate text using the OpenAI client.
+
+```typescript {7, 15-19}
+import { generateText } from 'ai';
+import { serve } from "@upstash/workflow/nextjs";
+import { WorkflowAbort } from '@upstash/workflow';
+import { createWorkflowOpenAI } from './model';
+
+export const { POST } = serve(async (context) => {
+  const openai = createWorkflowOpenAI(context);
+
+  // Important: Must get prompt in a step before using generateText 
```

Comment:
something like `must have a step before generateText`, it can be getting prompt or any other step